core/src/main/java/org/springframework/ldap/core/support/AbstractContextSource.java

```diff
@@ -119,10 +117,16 @@
 
 	public static final String SUN_LDAP_POOLING_FLAG = "com.sun.jndi.ldap.connect.pool";
 
-	private static final String JDK_142 = "1.4.2";
-
 	private DirContextAuthenticationStrategy authenticationStrategy = new SimpleDirContextAuthenticationStrategy();
 
+	public AbstractContextSource() {
+		try {
+			this.contextFactory = Class.forName("com.sun.jndi.ldap.LdapCtxFactory");
+		} catch (ClassNotFoundException e) {
+			LOG.error("Cannot resolve com.sun.jndi.ldap.LdapCtxFactory. contextFactory must be explicitly specified", e);
```

Comment:
This isn't necessarily an error since the application may call `setContextFactory` later. Would this work be better in the `afterPropertiesSet` method? Then, you would be able to state definitively that the class is correctly or incorrectly configured and throw an exception accordingly.

Reply:
I changed the logging level to `TRACE` and added the validation to the `afterPropertiesSet` method. I did not completely move the resolution to the `afterPropertiesSet` method since having it there would result in the same behavior change described [here](https://github.com/spring-projects/spring-ldap/pull/568#issuecomment-817032937).

Reply:
Could you elaborate on how the same behavior would result? It would be an error to use this class before calling `afterPropertiesSet`.

Reply:
elaborated below.